features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -1225,30 +1222,41 @@ void ATHandler::flush()
     }
 }
 
-void ATHandler::debug_print(const char *p, int len)
+void ATHandler::debug_print(const char *p, int len, ATType type)
 {
 #if MBED_CONF_CELLULAR_DEBUG_AT
     if (_debug_on) {
-#if MBED_CONF_MBED_TRACE_ENABLE
-        mbed_cellular_trace::mutex_wait();
-#endif
-        char c;
+        const int buf_size = len * 4; // x4 -> reserve space for extra characters
+        char *buffer = (char *)malloc(buf_size);
+        memset(buffer, 0, buf_size);
+
+        char *pbuf = buffer;
         for (ssize_t i = 0; i < len; i++) {
-            c = *p++;
-            if (!isprint(c)) {
-                if (c == '\r') {
-                    debug("\n");
-                } else if (c == '\n') {
-                } else {
-                    debug("#%02x", c);
-                }
+            const char c = *p++;
+            if (isprint(c)) {
+                *pbuf++ = c;
+            } else if (c == '\r') {
+                sprintf(pbuf, "<cr>");
```

Comment:
Thanks for the commit message. Made it really easy to tell why `\r` and `\n` were changed.